### ISSUE TITLE
[BUG] Compliance Policies not being added to Assignment report for new groups

### ISSUE THREAD

**Issue by gmenziesint** [almenscorner/IntuneCD#201]
**Describe the bug**
As per the title I can see that Compliance policies don't seem to be being added to the assignment report for new groups.

**To Reproduce**

Create a new group
Add a compliance policy to it and run the pipeline

**Expected behavior**
Compliance policies to be shown in the assignment report for groups

**Screenshots**

**Run type (please complete the following information):**
 - Mode: [e.g. 0 or 1] - 1
 - Client [e.g. Pipeline, local machine] - Pipeline
 - Version [e.g. 1.0.2] - 2.3.2

**Additional context**

**Comment by almenscorner** [comment]
I am unable to replicate this, I created a group and it was added to the assignment report successfully,

```json
     {
          "groupName": "201testgroup",
          "groupType": "StaticMembership",
          "membershipRule": null,
          "assignedTo": {
               "Compliance Policies": [
                    {
                         "name": "placeholder",
                         "type": "iosCompliancePolicy",
                         "intent": ""
                    }
               ]
          }
     },
```

Is the group included on the configuration in the backup?

**Comment by almenscorner** [comment]
I've been trying to replicate but no success so far.

Do you see the same issue if you run the backup locally and not in a pipeline?

**Comment by almenscorner** [comment]
I tried specifically with a Windows compliance policy as well but is was also added to the report,

```json
  {
       "name": "testwin",
       "type": "windows10CompliancePolicy",
       "intent": ""
  }
```

**Comment by almenscorner** [comment]
Can you provide the JSON for the policy not working so I can try with a replica?

**Comment by almenscorner** [comment]
I was able to replicate it (kind of) where not all groups were included in the report, it looks like it is because the loop exited early and did not go through each assignment. I've changed this in 2.3.3-beta2, try this version and see if the report is now correct: `pip3 install IntuneCD==2.3.3b2`

**Comment by almenscorner** [comment]
Fixed as of v2.3.3
